src/task/cluster.rs

```diff
@@ -497,6 +497,76 @@ impl Cluster {
     pub fn failover(&self) -> &Failover {
         &self.failover
     }
+
+    pub fn try_upgrade(mut self, new: &Cluster) -> Result<Self, GeninError> {
+        let old_hosts = self.hosts().first().unwrap().lower_level_hosts();
+        let new_hosts = new.hosts().first().unwrap().lower_level_hosts();
+
+        let mut diff = new_hosts
```

Comment:
I suppose `genin` should return error in case when new cluster does not contain old cluster's instances. Currently it will not raise the error.